MvvmCross/Core/Core/Navigation/MvxNavigationService.cs

```diff
@@ -25,10 +26,9 @@ public IMvxViewDispatcher ViewDispatcher
         }
 
         protected static readonly Dictionary<Regex, Type> Routes = new Dictionary<Regex, Type>();
-
         protected virtual IMvxNavigationCache NavigationCache { get; private set; }
-
-        public virtual IMvxViewModelLoader ViewModelLoader { get; set; }
+        protected IMvxViewModelLoader ViewModelLoader { get; set; }
+        protected ConditionalWeakTable<IMvxViewModel, TaskCompletionSource<object>> _tcsResults = new ConditionalWeakTable<IMvxViewModel, TaskCompletionSource<object>>();
```

Comment:
You only use `.Add` and `.TryGetValue` on `_tcsResults`. Does it mean you never remove items from it? Is that right? (never used ConditionalWeakTable before)